presto-orc/src/test/java/com/facebook/presto/orc/TestOrcRecordReader.java

```diff
@@ -0,0 +1,143 @@
+                        ImmutableList.of(8, 9),
+                        ImmutableList.of(
+                                "nested_col_0_double",
+                                "nested_col_1_boolean"
+                        )),
+                // 8
+                new OrcType(OrcType.OrcTypeKind.DOUBLE,
+                        ImmutableList.of(),
+                        ImmutableList.of()),
+                // 9
+                new OrcType(OrcType.OrcTypeKind.BOOLEAN,
+                        ImmutableList.of(),
+                        ImmutableList.of())
+        );
+        final Map<Integer, Type> includedColumnsGood1 = ImmutableMap.of(
+                0, BigintType.BIGINT,
+                1, new MapType(VarcharType.createVarcharType(2), DateType.DATE),
+                2, new ArrayType(DoubleType.DOUBLE)
+        );
+        final Map<Integer, Type> includedColumnsGood2 = ImmutableMap.of(
+                0, BigintType.BIGINT,
+                1, new MapType(VarcharType.createVarcharType(2), DateType.DATE),
+                2, new ArrayType(DoubleType.DOUBLE),
+                3, new RowType(ImmutableList.of(DoubleType.DOUBLE, BooleanType.BOOLEAN), Optional.of(ImmutableList.of("nested_col_0", "nested_col_1")))
+        );
+        final Map<Integer, Type> includedColumnsBad1 = ImmutableMap.of(
+                0, BooleanType.BOOLEAN,
+                1, new MapType(VarcharType.createVarcharType(2), DateType.DATE),
+                2, new ArrayType(DoubleType.DOUBLE)
+        );
+        final Map<Integer, Type> includedColumnsBad2 = ImmutableMap.of(
+                0, BigintType.BIGINT,
+                1, new ArrayType(DoubleType.DOUBLE),
+                2, new MapType(VarcharType.createVarcharType(2), DateType.DATE)
+        );
+        final Map<Integer, Type> includedColumnsBad3 = ImmutableMap.of(
+                0, BigintType.BIGINT,
+                1, new ArrayType(DoubleType.DOUBLE),
+                2, new MapType(DateType.DATE, VarcharType.createVarcharType(2))
+        );
+        final Map<Integer, Type> includedColumnsBad4 = ImmutableMap.of(
+                0, BigintType.BIGINT,
+                1, new MapType(VarcharType.createVarcharType(2), DateType.DATE),
+                2, new ArrayType(DoubleType.DOUBLE),
+                3, new RowType(ImmutableList.of(DoubleType.DOUBLE), Optional.of(ImmutableList.of("nested_col_0")))
+        );
+        final Map<Integer, Type> includedColumnsBad5 = ImmutableMap.of(
+                0, BigintType.BIGINT,
+                1, new MapType(VarcharType.createVarcharType(2), DateType.DATE),
+                2, new ArrayType(DoubleType.DOUBLE),
+                3, new RowType(ImmutableList.of(BigintType.BIGINT, BooleanType.BOOLEAN), Optional.of(ImmutableList.of("nested_col_0", "nested_col_1")))
+        );
+
+        assertEquals(OrcRecordReader.createStreamReaders(TestCachingOrcDataSource.FakeOrcDataSource.INSTANCE, types, null, includedColumnsGood1).length, 4);
```

Comment:
It seems unnecessary to assert that size of the return value. Just do `OrcRecordReader.createStreamReaders(TestCachingOrcDataSource.FakeOrcDataSource.INSTANCE, types, null, includedColumnsGood1)`